### PR TITLE
Unwind some Linq in a relatively hot path

### PR DESCRIPTION
**Pull request by Pilchie**

### CODE REVIEW

**Comment by Pilchie** [comment]
Tag @dotnet/roslyn-ide @dotnet/roslyn-perf

File: src/Workspaces/Core/Portable/Shared/Extensions/SymbolInfoExtensions.cs [hunk 1]
```diff
@@ -31,7 +31,9 @@ private static IEnumerable<ISymbol> GetAllSymbolsWorker(this SymbolInfo info)
 
         public static ISymbol GetAnySymbol(this SymbolInfo info)
         {
-            return info.GetAllSymbols().FirstOrDefault();
+            return info.Symbol != null
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Shared/Extensions/SymbolInfoExtensions.cs]: Comment as to why? And are there enough other callers to GetAllSymbols that should be looked at too?

Comment (AdamSpeight2008) [inline src/Workspaces/Core/Portable/Shared/Extensions/SymbolInfoExtensions.cs]: This changes the return value to be a `Boolean` not an `ISymbol`.
As what it the `info.Symbol == null`

Comment (Pilchie) [inline src/Workspaces/Core/Portable/Shared/Extensions/SymbolInfoExtensions.cs]: No, it uses the conditional operator on the next line to return items.

**Comment by CyrusNajmabadi** [comment]
What's the code scenario where these allocations show up? (just for my edification).  Thanks!

**Comment by Pilchie** [comment]
It's armchair optimization definitely.  I didn't see it in a trace, just looking at the number of callers in features like completion, classification, etc.

It seemed a little silly to create an enumerator state machine, and hash set (for the Distinct call) just to call .First on it when info.Symbol will almost always be non-null.

**Comment by CyrusNajmabadi** [comment]
That’s reasonable.  It just makes me ☹ that simple and clear code can’t also be extremely efficient and low allocating.

               -- Cyrus

**Comment by Pilchie** [comment]
In general I agree, but in this case I disagree that its harder to read...

**Comment by Pilchie** [comment]
Ping @dotnet/roslyn-ide.  Anyone?

**Comment by balajikris** [comment]
:+1:

**Comment by Pilchie** [comment]
Ping @CyrusNajmabadi/ @jasonmalinowski Can I get a signoff here?

**Comment by crummel** [comment]
Build triggered. sha1 is merged.

**Comment by CyrusNajmabadi** [comment]
:+1:

**Pull request merged**